props/jetpack_Oli_buttons.h

```diff
@@ -0,0 +1,945 @@
+        SLEEP(SaberBase::sound_length * 1000 + 10);
+        if (!flight_) { // Perform "Mando Talk" if not in flight mode!
+          // create animation for OLED of mando & boba talking
+          SaberBase::DoEffect(EFFECT_MANDOTALK,0);
+          SLEEP(SaberBase::sound_length * 1000 + 10);
+          PVLOG_NORMAL << "Mando: Nice shot!\nBoba: I was aiming for the other one!\n";
+        }
+        // create animation for OLED of viewfinder going back up
+        SaberBase::DoEffect(EFFECT_DISARM,0);
+        SLEEP(SaberBase::sound_length * 1000);
+        PVLOG_NORMAL << "Missile Launch Sequence Completed!\n";
+        missilelaunch_ = false;
+        }
+      YIELD();  // this means "sleep one loop"
+      }
+    STATE_MACHINE_END();
+  }
+
+  // Handles Jetpack mishaps events
+  void JetpackMishaps() {
+    if (jetpack_wav_player_.isPlaying()) {
+      return; // Wait for the current sound to finish
+    }
+    if (flight_) {
+      // Jetpack flying, play stuttering sound
+      jetpack_wav_player_.StopFromReader();     // Stop flight sound
+      SaberBase::DoEffect(EFFECT_STUTTERING,0); // Play stuttering sound
+    } else if (idle_) {
+      // Jetpack idle, play false start and reset idle timer
+      jetpack_wav_player_.StopFromReader();     // Stop idle sound
+      SaberBase::DoEffect(EFFECT_FALSESTART,0); // Play false start sound
+      timer_ = millis(); // Reset idle timer
+    } else ChooseRandomEffect(); // Play random mishap effect
+  }
+
+  // Jetpack off, play self destruct, meltdown or dud (randomly between the three)
+  void ChooseRandomEffect() {
+    int choice = random(3); // Generate a random number: 0, 1, or 2
+    switch (choice) {
+      case 0:  SaberBase::DoEffect(EFFECT_SELFDESTRUCT,0); break;
+      case 1:  SaberBase::DoEffect(EFFECT_MELTDOWN,0);     break;
+      case 2:  SaberBase::DoEffect(EFFECT_DUD,0);          break;
+      default: SaberBase::DoEffect(EFFECT_SELFDESTRUCT,0); break; // <-- Is a "default" even needed ???
+    }
+  }
+
+  // Initial start, transition to Idle Mode (from 0 to idle)
+  void StartIdleMode() {
+    if (jetpack_wav_player_.isPlaying()) {
+      return; // Wait for the current sound to finish
+    }
+    SaberBase::DoEffect(EFFECT_STARTIDLEMODE, 0); // start transition to idle mode
+    flight_ = false;
+    idle_   = true;
+    timer_  = millis(); // Reset idle timer
+    FastOn();
+    PVLOG_STATUS << "Jetpack Idling\n";
+  }
+
+  // Transition to Flying Mode (from idle to flying)
+  void StartFlightMode() {
+    if (jetpack_wav_player_.isPlaying()) {
+      return; // Wait for the current sound to finish
+    }
+    SaberBase::DoEffect(EFFECT_STARTFLIGHTMODE, 0); // start transition to flight mode
+    flight_ = true;
+    idle_   = false;
+    PVLOG_STATUS << "Jetpack Flying\n";
+  }
+
+  // Transition from Flying Mode (from flying to idle)
+  void StopFlightMode() {
+    if (jetpack_wav_player_.isPlaying()) {
+      return; // Wait for the current sound to finish

```

Comment:
This comment is misleading, as there is no waiting involved.
This just disables this function if something else is happening, which seems like it could lead to bad user experiences.
(I want to stop, why isn't it working???)